Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+
+=== Adding usage text
+
+Try and run `./bin-wrappers/git psuh -h`. Your command should crash at the end.
+That's because `-h` is a special case which your command should handle by
+printing usage.
+
+Take a look at `Documentation/technical/api-parse-options.txt`. This is a handy
+tool for pulling out options you need to be able to handle, and it takes a
+usage string.
+
+In order to use it, we'll need to prepare a NULL-terminated usage string and a
+builtin_psuh_options array. Add a line to `#include "parse-options.h"`.
+
+At global scope, add your usage:
+
+----
+static const char * const psuh_usage[] = {
+	N_("git psuh"),
+	NULL,
+};
+----
+
+Then, within your cmd_psuh implementation, we can declare and populate our
+`option` struct. Ours is pretty boring but you can add more to it if you like:
+
+----
+	struct option options[] = {
+		OPT_END()
+	};
+----
+
+Finally, before you print your args and prefix, add the call to
+`parse-options()`:
+
+----
+	argc = parse_options(argc, argv, prefix, options, psuh_usage, 0);
+----
+
+Build again. Now, when you run with -h, you should see your usage printed and
+your command terminated before anything else interesting happens. Great!
+
+Go ahead and commit this one, too.
+
+== Testing
+
+It's important to test your code - even for a little toy command like this one.
+So let's add some tests.
+
+Related reading: `t/README`
+
+=== Overview of Testing Structure
+
+The tests in Git live in t/ and are named with a 4-decimal digit, according to
+the schema shown in the Naming Tests section of `t/README`.
+
+=== Writing Your Test
+
+Since this a toy command, let's go ahead and name the test with t9999. However,
+as many of the family/subcmd combinations are full, best practice seems to be
+to find a command close enough to the one you've added and share its naming
+space.
+
+Create your test script and mark it executable:
+
+----
+touch t/t9999-psuh-codelab.sh
+chmod +x t/t9999-psuh-codelab.sh
+----
+
+Begin with the header as so (see
+"Writing Tests" and "Source 'test-lib.sh'" in `t/README`):
+
+----
+#!/bin/sh
+
+test_description='git-psuh test
+
+This test runs git-psuh and makes sure it does not crash.'
+
+. ./test-lib.sh
+----
+
+Tests are framed inside of a `test_expect_success` in order to output TAP
+formatted results. Begin your first test and set up the repo to test in:
+
+----
+test_expect_success 'runs correctly with no args' '
+	rm -rf workbench upstream &&
+	test_create_repo upstream &&
+----
+
+`test_create_repo` comes from `test-lib.sh`. Next, we'll modify the above to
+move into the new repo and run our new command:
+
+----
+test_expect_success 'runs correctly with no args' '
+	rm -rf workbench upstream &&
+	test_create_repo upstream &&
+	(
+		cd upstream &&
+		git psuh
+	)
+'
+----
+
+Indicate that you've run everything you wanted by adding the following at the
+bottom of your script:
+
+----
+test_done
+----
+
+=== Running Locally
+
+Let's try and run locally:
+
+----
+make -j16
```

Comment:
I recommended the same for years, until somebody pointed out hat `make -j$(nproc)` makes more sense...

Reply:
Nice, that's handy.